web/src/app/modules/views/main/editors/modules/tool-pallette/util/graphTransformer.ts

```diff
@@ -0,0 +1,134 @@
+        return this.dataService.readContents(this.parent.url, true).then( (content: IContainer[]) => {
+            // Delete Connections
+            let chain = Promise.resolve();
+            content.forEach(elem => {
+                if (this.elementProvider.isConnection(elem)) {
+                    let currentConnection: IModelConnection = elem as IModelConnection;
+                    if (currentConnection.source.url === node.url || currentConnection.target.url === node.url) {
+                        chain = chain.then(() => this.deleteConnection(currentConnection, compoundId));
+                    }
+                }
+            });
+            return chain;
+        }).then(() => this.dataService.deleteElement(node.url, true, compoundId));
+    }
+
+    private deleteConnection(connection: IModelConnection, compoundId: string): Promise<void> {
+        return this.removeConnectionFromSource(connection, compoundId)
+            .then(() => this.removeConnectionFromTarget(connection, compoundId))
+            .then(() => this.dataService.deleteElement(connection.url, true, compoundId));
+    }
+
+    private removeConnectionFromSource(connection: IModelConnection, compoundId: string): Promise<void> {
+        return this.dataService.readElement(connection.source.url, true)
+            .then((source: IModelNode) => {
+                let proxyToDelete: Proxy = source.outgoingConnections.find((proxy: Proxy) => proxy.url === connection.url);
+                Arrays.remove(source.outgoingConnections, proxyToDelete);
+                return source;
+            })
+            .then((source: IContainer) => this.dataService.updateElement(source, true, compoundId));
+    }
+
+    private removeConnectionFromTarget(connection: IModelConnection, compoundId: string): Promise<void> {
+        return this.dataService.readElement(connection.target.url, true)
+            .then((target: IModelNode) => {
+                let proxyToDelete: Proxy = target.incomingConnections.find((proxy: Proxy) => proxy.url === connection.url);
+                Arrays.remove(target.incomingConnections, proxyToDelete);
+                return target;
+            })
+            .then((target: IContainer) => this.dataService.updateElement(target, true, compoundId));
+    }
+
+    // Create
+    public createSubgraph(templates: IContainer[]): Promise<void> {
+        let compundId = Id.uuid;
+        let urlMap: {[old: string]: IModelNode} = {};
+        // Old URL -> New Node map
+
+        let chain: Promise<any> = Promise.resolve();
+
+        for (const template of templates) {
+            if (this.elementProvider.isNode(template)) {
+                let temp = <IModelNode> template;
+                let newCoord = { x: temp.x, y: temp.y + 100};
+                let factory = GraphElementFactorySelector.getNodeFactory(template, newCoord, this.dataService);
+                chain = chain.then(() => factory.create(this.parent, false))
```

Comment:
In this case, I think it would be beneficial not to chain the promises manually, but use async/await to make the code more readable.